x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchActionIT.java

```diff
@@ -456,13 +456,34 @@ public void testUpdateRunningKeepAlive() throws Exception {
             response.decRef();
         }
 
-        final AsyncSearchResponse response2 = getAsyncSearch(response.getId(), TimeValue.timeValueMillis(1));
         try {
-            assertThat(response2.getExpirationTime(), lessThan(expirationTime));
-            ensureTaskNotRunning(response2.getId());
-            ensureTaskRemoval(response2.getId());
+            if (randomBoolean()) {
+                final AsyncSearchResponse response2 = getAsyncSearch(response.getId(), TimeValue.timeValueMillis(1));
+                try {
+                    assertThat(response2.getExpirationTime(), lessThan(expirationTime));
+                    ensureTaskNotRunning(response2.getId());
+                    ensureTaskRemoval(response2.getId());
+                } finally {
+                    response2.decRef();
+                }
+            } else {
+                try {
+                    AsyncStatusResponse statusResponse = getAsyncStatus(response.getId(), TimeValue.timeValueMillis(5));
```

Comment:
The async status code path is more prone to hit the race condition where the results have disappeared due to expiration than the async search results code path (not totally sure why - I just think a few more instructions of code have to execute due to separate code paths), so I increased the expiration time to 5ms rather than 1ms. On my local testing this causes the non-exception path to happen more frequently than the exception path, but both are valid test results.